src/core/qgspallabeling.cpp

```diff
@@ -3706,6 +3764,69 @@ void QgsPalLabeling::dataDefinedTextBuffer( QgsPalLayerSettings &tmpLyr,
   }
 }
 
+void QgsPalLabeling::dataDefinedTextMask( QgsPalLayerSettings &tmpLyr,
+    const QMap< QgsPalLayerSettings::Property, QVariant > &ddValues )
+{
+  QgsTextMaskSettings mask = tmpLyr.format().mask();
```

Comment:
Again, could return early if there's the datadefined properties has no active properties.

Reply:
You mean ddValues.isEmpty() ?